repository_service_tuf_worker/repository.py

```diff
@@ -398,61 +417,128 @@ def bootstrap(
 
         return asdict(result)
 
-    def add_targets(self, payload: Dict[str, Any], update_state: str) -> None:
+    def publish_targets(self):
         """
-        Updates 'bins' roles metadata, assigning each passed target to the
-        correct bin.
+        Publish targets from SQL DB as persistent TUF Metadata in the backend
+        storage.
+        """
+
+        # lock to avoid race conditions
+        with self._redis.lock("publish_targets", timeout=5.0):
+            # get all delegated role names with unpublished targets
+            unpublished_roles = targets_crud.read_unpublished_rolenames(
+                self._db
+            )
 
-        Assignment is based on the hash prefix of the target file path. All
-        metadata is signed and persisted using the configured key and storage
-        services.
+            if len(unpublished_roles) == 0:
+                logging.debug("No new targets in delegated roles. Finishing")
+                return None
+
+            # initialize the new snapshot targets meta and published targets
+            # from DB SQL
+            new_snapshot_targets_meta: List[Tuple(str, int)] = []
```

Comment:
I think you can shorten your variable name here with `new_snapshot_meta` as by definition the TUF spec snapshot meta contains information about all target metadata files.